pcsx2/x86/iR3000A.cpp

```diff
@@ -866,11 +866,22 @@ static __fi void recClearIOP(u32 Addr, u32 Size)
 		pc += PSXREC_CLEARM(pc);
 }
 
+// The x86 emitter doesn't like xFastCall(DevCon.Warning("text")) on linux
+__fi void recNotifyBranchToZero()
+{
+	DevCon.Warning("[R3000 Recompiler] Warning: Branch to 0x0!");
+}
+
 void psxSetBranchReg(u32 reg)
 {
 	psxbranch = 1;
 
 	if( reg != 0xffffffff ) {
+
+		if (PSX_IS_CONST1(reg) ? g_psxConstRegs[reg] == 0 : psxRegs.GPR.r[reg] == 0)
+			// We are still gonna hard crash and burn but at least the emulog tells us why :)
+			xFastCall((void*)recNotifyBranchToZero);
```

Comment:
Is there any reg flushing function on the IOP?  I'm nervous of calling stuff outside the JIT without flushing, that can be a bit dangerous

Reply:
I'll double check, it's hard to test because the recompiler is going to die anyways :p